tests/Microsoft.Identity.Test.Unit.net45/UtilTests/ScopeHelperTests.cs

```diff
@@ -0,0 +1,52 @@
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            TestCommon.ResetInternalStaticCaches();
+        }
+
+        [TestMethod]
+        public void ScopeHelperPerf()
+        {
+            ISet<string> scopeSet = null;
+            using (new PerformanceValidator(100, "Convert scope string to set"))
+            {
+                // about 500ms for 5000 iterations -> down to 160ms after replacing SortedSet with HashSet
+                for (int i = 0; i < 1000; i++)
+                {
+                    scopeSet = ScopeHelper.ConvertStringToScopeSet(LotsOfScopes);
+                }
+            }
+            bool contains = true;
+            using (new PerformanceValidator(100, "Scope contains"))
+            {
+                // about 150ms for 10000 iterations -> down to 5ms after replacing SortedSet with HashSet
```

Comment:
the scopes are now a HashSet, so lookups are mostly O(1), hence the super low 5ms

Reply:
awesome.

---
In reply to: [472266412](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2000#discussion_r472266412) [](ancestors = 472266412)